### ISSUE TITLE
Thin bug with substitution

### ISSUE THREAD

**Issue by cser** [Proektsoftbg/Calcpad#42]
So, this is a problem:
```
#def ____$ = &emsp;&emsp;
#def EQU_M$(equ$; macro$)
    #hide
    macro$$
    #show
    '<p align="center">'equ$',</p>
    macro$$
#end def

'This shows x²+y²+z²=10²+20²+0²=1400
#def VARS$
    'where'x = 10
    '____$'y = 20
    '____$'z = 30
#end def
EQU_M$(x^2 + y^2 + z^2; VARS)
```
It shows 0² instead of 30², but result is correct. Generated code is correct too and works normally. It seems that it's a problem with calling macros inside macros, may be output is correct because it shows after **2 times** replacer calling but substitution is processed after 1-st time.

**Comment by cser** [comment]
But why it works so only for last variable? - No matter how much variables are.

**Comment by cser** [comment]
@Proektsoftbg, Sorry, it's not a bug with macroses.
```
#hide
x = 10
y = 20
z = 30
#show
x^2 + y^2 + z^2'shows x²+y²+z²=10²+20²+0²=1400 (0² instead of 30²)
```

**Comment by Proektsoftbg** [comment]
@cser Thank you for reporting that! This is a very wierd bug. I will have a look.

**Comment by Proektsoftbg** [comment]
Hi @cser, I fixed this and ammended the last release 5.8.2.

**Comment by cser** [comment]
Yes, now it's OK. Seems that other isn't broken.
